File: 6.Function/3040.c
#include <stdio.h>
#include <stdlib.h>

char tg[] = "癸甲乙丙丁戊己庚辛壬";
char dz[] = "亥子丑寅卯辰巳午未申酉戌";

void YEAR(int n, int arr[]) {
    for (int i = 0; i < n; i++) {
        int curr = arr[i];
        int tgi = ((curr - 3) % 10) * 3;
        int dzi = ((curr - 3) % 12) * 3;

        printf("%d = %c%c%c%c%c%c年\n", curr, tg[tgi], tg[tgi+1], tg[tgi+2], dz[dzi], dz[dzi+1], dz[dzi+2]);
    }
}

int main() {

    int years[1024], c = 0, num;

    while (scanf("%d", &num) == 1) {
        years[c] = num;
        c++;
    }

    YEAR(c, years);

    return 0;
}

File: 6.Function/3040.py

tg = ['甲','乙','丙','丁','戊','己','庚','辛','壬','癸']
dz = ['子','丑','寅','卯','辰','巳','午','未','申','酉','戌','亥']

def YEAR(lst):

    for l in lst:

        year = int(l)

        tgi = (year - 3) % 10 - 1
        dzi = (year - 3) % 12 - 1

        print("{} = {}{}年".format(year, tg[tgi], dz[dzi]))

years = []

while True:
    year = input()

    if year == "q":
        break

    years.append(year)

YEAR(years)
    
